version: '3.1'

services:
  
  task-manager:
    build:
      context: ./task-manager
      dockerfile: ./Dockerfile
    restart: on-failure
    ports:
      - "5000:5000"
    environment:
      CURRENT_ENV: docker
    depends_on:
      - mongo
      - minio
      - rabbitmq

  subtask-manager:
    build:
      context: ./subtask-manager
      dockerfile: ./Dockerfile
    restart: on-failure
    ports:
      - "5001:5001"
    environment:
      CURRENT_ENV: docker
    depends_on:
      - mongo
      - minio
      - rabbitmq

  mongo:
    image: mongo
    ports:
      - "27017:27017"
    environment:
      MONGO_INITDB_ROOT_USERNAME: root
      MONGO_INITDB_ROOT_PASSWORD: root
    volumes:
      - mongodbdata:/data/db

  mongo-express:
    image: mongo-express
    ports:
      - "8081:8081"
    environment:
      ME_CONFIG_MONGODB_ADMINUSERNAME: root
      ME_CONFIG_MONGODB_ADMINPASSWORD: root
      ME_CONFIG_MONGODB_URL: "mongodb://root:root@mongo:27017"
    depends_on:
      - mongo

  minio:
    image: bitnami/minio
    ports:
      - "9000:9000"
      - "9001:9001"
    environment:
      MINIO_ROOT_USER: root
      MINIO_ROOT_PASSWORD: root-root
      MINIO_DEFAULT_BUCKETS: images
    volumes:
      - miniodata:/data

  rabbitmq:
    image: rabbitmq:3-management-alpine
    container_name: "rabbitmq"
    ports:
      - "5672:5672"
      - "15672:15672"
    volumes:
      - rabbitmq-data:/var/lib/rabbitmq/
      - rabbitmq-log:/var/log/rabbitmq

volumes:
  mongodbdata:
  miniodata:
  rabbitmq-data:
  rabbitmq-log:
